### ISSUE TITLE
ALMNH Mamm - Contact

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/data-migration#246]
Hi Teresa,

 

I have attached a template I created for our Mammals collection that I need your help on.  It’s pretty much standard, except for occasional specimen remarks.  I thought if you could help me set up the fields, I could test the upload and adapt it to the rest of the collection. 

 

I probably don’t have the correct field names as there are a lot of disclaimers about field names in the “how to” document.

 

I looked at all the bulkload templates, but there is just too much to remember and I can’t print out anything here at home.

 

Please help me when you get the chance!  It shouldn’t be too daunting as we don’t have a ton of information, just specimen measurements (which, I might add, were never entered when the datasheets were first entered into our Filemaker database (!)  I will need to enter about 1500 more measurements and would like to have the exact fields in the template so I can use them to record the data from the data sheets.

Got that?!

**Comment by Jegelewicz** [comment]
> Mary Beth,
> 
> OK, I had a few minutes so I did a few things to the file.
> 
> First - when using the main bulkloader, you cannot have more than one row per record (catalog number), the skin and skeleton need to be on the same line, so I added a second set of "part" fields.
> 
> Second - when using the main bulkloader you cannot bulkload part attributes, these will have to get loaded later using the part attribute bulkload tool. I moved those things to a second sheet so you can see how those will be loaded. The challenge here is that you need the GUID to load them. If you are letting Arctos assign catalog numbers, we will have to download the records and use the Other ID (ZV numbers) to match the parts with their attributes. It can be done, just will take a bit of work.
> 
> Third - the individual ZV numbers for the parts will probably be better off as part attributes - you can see how I added them to the part attribute bulkload file. THEN to help keep everything together, add only one ZV number as an Other ID, leaving off the numbers after the last period. so in the record you will have an original identifier of ZV1987.0003.000, the skin will have an Other ID of ZV1987.0003.0001.0001 and the skeleton will have an Other ID attribute of ZV1987.0003.0001.0002. Hope that makes sense!
> 
> Fourth - Each attribute has to be entered separately - see how total length, tail length, etc are now formatted. Please note that all of these attributes also require a date and a determiner. I generally use the preparator name in the determiner field. If there is no preparator given, I use the collector. If there is no collector, use "unknown". For the date, I use the preparation date, if no prep date, then collection date, if no collection date, then put today. Given all of this, you don't need to have "(TL) = 110 mm (TV) = 40 mm (HF) = 27 mm (E) = 14 mm" in Collection Object Remarks unless you really want everything in the record twice.
> 
> Fifth - There are a few required fields that you didn't have in the file - I added them way out to the right. I typically fill the Event Assigned by Agent and Event Assigned Date with the collector name and date of collection. The Begin and Ended Dates should be the verbatim date converted to ISO format (YYYY-MM-DD). If your verbatim date is unknown, you can either enter a kind of nonsense date (1900-01-01 is a favorite) or make some guesses (it wasn;t collected before the collector was born and it definitely wasn't collected after "today"). Just be consistent in your treatment so you can figure stuff out later.
> 
> Sixth - I put a formula in the VERBATIM_LOCALITY and HIGHER_GEOG fields - gives you an easy way to get those fields filled in.
> 
> Whew! I think that should explain everything - although I am pretty sure there is something I forgot to say here. Just let me know if you have questions.
> 
> Adios,
> 

Teresa
[ALMNH Mamm Bulkload.xlsx](https://github.com/ArctosDB/data-migration/files/4453460/ALMNH.Mamm.Bulkload.xlsx)

**Comment by Jegelewicz** [comment]
@mbprondzinski I also forgot to mention that you will need to go through the same steps we did for paleo. All of the Agents, Accessions, and Taxon names need to be in Arctos before you will be able to load the records! I've created the [project in GitHub](https://github.com/ArctosDB/data-migration/projects/26) will all of the steps you should follow.

**Comment by mbprondzinski** [comment]
All right, thanks for all this Teresa.  I plan on entering the original Taxon names, despite the nomenclature changes, just so I have a record of the original id, is that okay?

**Comment by Jegelewicz** [comment]
>  I plan on entering the original Taxon names, despite the nomenclature changes, just so I have a record of the original id, is that okay?

Absolutely! I consider that a best practice. Still, we will need to conform those names to agree with the code taxonomy code table or change them to the A {string} using some conforming taxon.

**Comment by Jegelewicz** [comment]
> Teresa,
> 
> Thanks for getting to this so quickly!  Here are my responses and your explanations seem pretty straight forward.
> 
> From: Teresa Mayfield <teresa.jegelewicz@jegelewicz.net>
> Sent: Wednesday, April 8, 2020 7:50 PM
> To: Prondzinski, Mary Beth <mbprondzinski@ua.edu>
> Subject: Re: Mammals Bulkload Template

Mary Beth,

OK, I had a few minutes so I did a few things to the file.

First - when using the main bulkloader, you cannot have more than one row per record (catalog number), the skin and skeleton need to be on the same line, so I added a second set of "part" fields.

> I figured that much…thanks!

Second - when using the main bulkloader you cannot bulkload part attributes, these will have to get loaded later using the part attribute bulkload tool. I moved those things to a second sheet so you can see how those will be loaded. The challenge here is that you need the GUID to load them. If you are letting Arctos assign catalog numbers, we will have to download the records and use the Other ID (ZV numbers) to match the parts with their attributes. It can be done, just will take a bit of work.

> With this collection, I can easily assign Catalog numbers if that is easiest.

Third - the individual ZV numbers for the parts will probably be better off as part attributes - you can see how I added them to the part attribute bulkload file. THEN to help keep everything together, add only one ZV number as an Other ID, leaving off the numbers after the last period. so in the record you will have an original identifier of ZV1987.0003.000, the skin will have an Other ID of ZV1987.0003.0001.0001 and the skeleton will have an Other ID attribute of ZV1987.0003.0001.0002. Hope that makes sense!

> Yes, that’s how I would have done it as well.  My only concern was if someone searched for the actual number with the 001 or 002, they wouldn’t find it.

Fourth - Each attribute has to be entered separately - see how total length, tail length, etc are now formatted. Please note that all of these attributes also require a date and a determiner. I generally use the preparator name in the determiner field. If there is no preparator given, I use the collector. If there is no collector, use "unknown". For the date, I use the preparation date, if no prep date, then collection date, if no collection date, then put today. Given all of this, you don't need to have "(TL) = 110 mm (TV) = 40 mm (HF) = 27 mm (E) = 14 mm" in Collection Object Remarks unless you really want everything in the record twice.

> Okay.  That was the field where the measurements were recorded because our database had no place specific to record them.  I don’t intend to keep it in the remarks field once I get it parsed into the correct attribute fields.  Is there an easy way for me to parse those into their respective fields?  I can probably figure it out, but if you know of an easy way, please let me know!

Fifth - There are a few required fields that you didn't have in the file - I added them way out to the right. I typically fill the Event Assigned by Agent and Event Assigned Date with the collector name and date of collection. The Begin and Ended Dates should be the verbatim date converted to ISO format (YYYY-MM-DD). If your verbatim date is unknown, you can either enter a kind of nonsense date (1900-01-01 is a favorite) or make some guesses (it wasn;t collected before the collector was born and it definitely wasn't collected after "today"). Just be consistent in your treatment so you can figure stuff out later.

> Thanks for adding those.  I knew I needed date fields for a lot of stuff, but since this particular record had nothing recorded, I did not add them.

Sixth - I put a formula in the VERBATIM_LOCALITY and HIGHER_GEOG fields - gives you an easy way to get those fields filled in.

> Thanks!

Whew! I think that should explain everything - although I am pretty sure there is something I forgot to say here. Just let me know if you have questions.

**Comment by Jegelewicz** [comment]
> With this collection, I can easily assign Catalog numbers if that is easiest.

That will make things easier!

> Yes, that’s how I would have done it as well. My only concern was if someone searched for the actual number with the 001 or 002, they wouldn’t find it.

You can add all of the numbers as original identifiers, but I think that might be overkill? We might need to think about an appropriate message when a search by Other Identifier comes up with no results. A message like "you may also want to search by Part Attribute = Other ID". might help alleviate that problem.

**Comment by Jegelewicz** [comment]
> Okay. That was the field where the measurements were recorded because our database had no place specific to record them. I don’t intend to keep it in the remarks field once I get it parsed into the correct attribute fields. Is there an easy way for me to parse those into their respective fields? I can probably figure it out, but if you know of an easy way, please let me know!

I do have an easy way to do that. All I need is a file that has the Arctos catalog number and the remark - I can easily do the rest with some formulas in Excel. Then we upload the attributes and get help from Dusty to remove the remark. Alternatively, you can just upload the records without the remark. As soon as the records are there we can upload the attributes.

**Comment by mbprondzinski** [comment]
Good deal!  Thanks…
